src/main/java/io/jawg/osmcontributor/ui/activities/PhotoActivity.java

```diff
@@ -388,43 +399,83 @@ public void onUploadFinishedEVent(final String photoId) {
         flickrPhotoClient.setProperties(oauthRequest.getParams(), new Callback<String>() {
             @Override
             public void success(String s, Response response) {
-                setPhotoTag(photoId);
+                try {
+                    setPhotoTag(photoId);
+                } catch (UnsupportedEncodingException e) {
```

Comment:
Either wrap encoding exception and throw an IllegalStateException and Log the error, or redirect to failure();